tla-types/src/test/scala/at/forsyte/apalache/tla/typecheck/etc/TestTypeCheckerTool.scala

```diff
@@ -43,8 +44,18 @@ class TestTypeCheckerTool extends AnyFunSuite with BeforeAndAfterEach with EasyM
 
   def loadSpecFromResource(name: String): Source = {
     // Previously, we were using fromResource, but it was too unstable across environments
-    // (e.g., it failed in Intellij Idea). Now we are just reading it from the current working directory.
-    Source.fromFile(s"tla-types/src/test/resources/$name.tla")
+    // (e.g., it failed in Intellij Idea). Now we are just reading it from $APALACHE_HOME/tla-types/src/test/resources.
+    // This is consistent with the behavior of SanyImporter when it is run in tests.
+    System.getenv("APALACHE_HOME") match {
+      // Warn if environment variable APALACHE_HOME is not set
+      case null =>
+        logger.error("Not running from fat JAR and APALACHE_HOME is not set.")
+        logger.error("Set APALACHE_HOME to a directory where Apalache has been checked out.")
+        throw new IllegalStateException("Missing APALACHE_HOME to run the tests")
+
+      case apalacheHome: String =>
+        Source.fromFile(s"$apalacheHome/tla-types/src/test/resources/$name.tla")
+    }
```

Comment:
I get complaints about unrelated changes added into PRs, so I'm going to start complaining about unrelated changes added to other people's PRs. Consider creating a separate issue for this and factoring this into a separate PR.

Reply:
It is related, as I had to run the tests, and they were not runnable in Idea, as we had two different solutions to dealing with resources.